src/engines/Cesium/hooks/useCamera.ts

```diff
@@ -0,0 +1,154 @@
+
+    const c = getCamera(viewer);
+    if (c && !isEqual(c, camera)) {
+      emittedCamera.current.push(c);
+      // The state change is not sync now. This number is how many state updates can actually happen to be merged within one re-render.
+      if (emittedCamera.current.length > 10) {
+        emittedCamera.current.shift();
+      }
+      onCameraChange?.(c);
+    }
+  }, [cesium, camera, onCameraChange]);
+
+  const handleCameraChange = useCallback(() => {
+    updateCamera();
+  }, [updateCamera]);
+
+  const handleCameraMoveEnd = useCallback(() => {
+    updateCamera();
+  }, [updateCamera]);
+
+  useEffect(() => {
+    if (camera && !emittedCamera.current.includes(camera)) {
+      engineAPI.flyTo(camera, { duration: 0 });
+      emittedCamera.current = [];
+    }
+  }, [camera, engineAPI]);
+
+  useEffect(() => {
+    if (!cesium.current?.cesiumElement) return;
+    const allowCameraMove = !!(featureFlags & FEATURE_FLAGS.CAMERA_MOVE);
+    const allowCameraZoom = !!(featureFlags & FEATURE_FLAGS.CAMERA_ZOOM);
+    const allowCameraTilt = !!(featureFlags & FEATURE_FLAGS.CAMERA_TILT);
+    const allowCameraLook = !!(featureFlags & FEATURE_FLAGS.CAMERA_LOOK);
+    cesium.current.cesiumElement.scene.screenSpaceCameraController.enableTranslate =
+      allowCameraMove;
+    cesium.current.cesiumElement.scene.screenSpaceCameraController.enableRotate = allowCameraMove;
+    cesium.current.cesiumElement.scene.screenSpaceCameraController.enableLook = allowCameraLook;
+    cesium.current.cesiumElement.scene.screenSpaceCameraController.enableTilt = allowCameraTilt;
+    cesium.current.cesiumElement.scene.screenSpaceCameraController.enableZoom = allowCameraZoom;
+  }, [cesium, featureFlags]);
+
+  // move to initial position at startup
+  const initialCameraFlight = useRef(false);
+
+  const handleMount = useCustomCompareCallback(
+    () => {
+      if (initialCameraFlight.current) return;
+      initialCameraFlight.current = true;
+      if (property?.camera?.limiter?.enabled && property?.camera?.limiter?.targetArea) {
+        engineAPI.flyTo(property?.camera?.limiter?.targetArea, { duration: 0 });
+      } else if (property?.camera?.camera) {
+        const camera = property?.camera?.camera;
+        engineAPI.flyTo(camera as Camera, { duration: 0 });
+      }
+      const camera = getCamera(cesium?.current?.cesiumElement);
+      if (camera) {
+        onCameraChange?.(camera);
+      }
+      onMount?.();
+    },
+    [
+      engineAPI,
+      property?.camera?.camera,
+      property?.camera?.limiter?.enabled,
+      onCameraChange,
+      onMount,
+    ],
+    (prevDeps, nextDeps) =>
+      prevDeps[0] === nextDeps[0] &&
+      isEqual(prevDeps[1], nextDeps[1]) &&
+      prevDeps[2] === nextDeps[2] &&
+      prevDeps[3] === nextDeps[3] &&
+      prevDeps[4] === nextDeps[4],
+  );
+
+  const handleUnmount = useCallback(() => {
+    initialCameraFlight.current = false;
+  }, []);
```

Comment:
These `handleMount` and `handleUnmount` are a function for Resium's onMount and onUnmount. The logic is related to the camera, but the function is for Resium's event. So you should expose the these API as `mountCamera` and `unmountCamera`, then use them in `Cesium/hooks.ts`

Reply:
Make sense. I refactor this part to another useMount file since useCamera needs some props from Cesium hooks and there's define order issue.